packages/core/src/shared/telemetry/util.ts

```diff
@@ -84,8 +84,8 @@ export function convertLegacy(value: unknown): boolean {
     }
 }
 
-export const getClientId = shared(
-    async (globalState: Memento, isTelemetryEnabled = new TelemetryConfig().isEnabled(), isTest?: false) => {
+export const getClientId = memoize(
```

Comment:
Necessary to make this synchronous because `getUserAgent` is used in constructors. memoize() better fits the semantics anyway

Reply:
Unrelated: Should we make significant setup in constructors a code smell? I've seen multiple cases where something eventually needs to be `async`. Maybe define a pattern to have a static `async create()` method for complex cases.

Reply:
> Maybe define a pattern to have a static `async create()` method for complex cases.

That's what I ended up doing for wizards: https://github.com/aws/aws-toolkit-vscode/commit/cf3f0fe020370ed80f55c7f4aba8f3691e8ce73c